flytekit/remote/remote.py

```diff
@@ -252,155 +248,152 @@ def register(self, entity):
-            # isn't any existing launch plan with a bound auth role to fall back on.
-            (
-                AuthRole(
-                    assumable_iam_role=auth_config.ASSUMABLE_IAM_ROLE.get(),
-                    kubernetes_service_account=auth_config.KUBERNETES_SERVICE_ACCOUNT.get(),
-                )
-                if auth_role is None
-                else auth_role
-            ),
-        )
+    def _(self, entity, inputs: typing.Dict[str, typing.Any], execution_name=None):
+        return self._execute(entity.id, inputs, execution_name)
 
     @execute.register
     @exception_scopes.system_entry_point
-    def _(
-        self,
-        entity: FlyteWorkflow,
-        inputs,
-        name=None,
-        notification_overrides=None,
-        label_overrides=None,
-        annotation_overrides=None,
-        auth_role=None,
-    ):
-        return self.execute(
-            _create_launch_plan(entity, auth_role),
-            inputs,
-            name,
-            notification_overrides,
-            label_overrides,
-            annotation_overrides,
-            auth_role,
-        )
-
-    @execute.register
-    @exception_scopes.system_entry_point
-    def _(
-        self,
-        entity: FlyteLaunchPlan,
-        inputs,
-        name=None,
-        notification_overrides=None,
-        label_overrides=None,
-        annotation_overrides=None,
-        auth_role=None,
-    ):
-        return _execute(
-            entity.id, inputs, name, notification_overrides, label_overrides, annotation_overrides, auth_role
-        )
+    def _(self, entity: FlyteWorkflow, inputs: typing.Dict[str, typing.Any], execution_name=None):
+        return self.execute(_create_launch_plan(entity, self.auth_role), inputs, execution_name)
 
     # Flytekit Entities
     # -----------------
 
     @execute.register
     @exception_scopes.system_entry_point
-    def _(self, entity: PythonFunctionTask):
-        pass
+    def _(self, entity: PythonTask, inputs: typing.Dict[str, typing.Any], execution_name: str = None):
+        try:
+            flyte_task: FlyteTask = self.fetch_task(self.project, self.domain, entity.name, self.version)
+        except Exception:
+            # TODO: fast register the task if fast=True
+            flyte_task: FlyteTask = self.register(entity)
+        return self.execute(flyte_task, inputs, execution_name)
 
     @execute.register
     @exception_scopes.system_entry_point
-    def _(self, entity: PythonFunctionWorkflow):
-        pass
-
-    @execute.register
-    @exception_scopes.system_entry_point
-    def _(self, entity: LaunchPlan):
-        pass
+    def _(self, entity: WorkflowBase, inputs: typing.Dict[str, typing.Any], execution_name=None):
+        try:
+            flyte_workflow: FlyteWorkflow = self.fetch_workflow(self.project, self.domain, entity.name, self.version)
+        except Exception:
+            flyte_workflow: FlyteWorkflow = self.register(entity)
```

Comment:
how does this work, shouldn't we have to register a default launch plan too and use that identifier when executing in line 388 below?